### PR TITLE
[Keychain] Prevent keychain access from prompting user for permissions on macOS

### PR DESCRIPTION
**Pull request by ncooke3**
### Context
This PR adds the [`kSecUseDataProtectionKeychain`](https://developer.apple.com/documentation/security/ksecusedataprotectionkeychain) search attribute to the keychain wrapper APIs in `GULKeychainStorage.h` and `GULKeychainUtils.h`.  The search key will make the macOS keychain behave like the iOS-style keychain. This is needed to avoid the keychain access pop-ups that occur when a macOS-style keychain is accessed. 

Permissions pop-up from `Keychain Access.app`

GoogleUtilities exposes two public headers that facilitate interaction with the Keychain: `GULKeychainStorage.h` and `GULKeychainUtils.h`. `GULKeychainStorage.h` implementation uses `GULKeychainUtils.h` to actually make call the Keychain's native APIs. Because of this, adding the `kSecUseDataProtectionKeychain` was, at minimum, required in `GULKeychainUtils.h`. But, to avoid future implementation changes where `GULKeychainStorage.h` no longer uses `GULKeychainUtils.h` as an implementation detail, I have ensured the key is inject at the `GULKeychainStorage.h` level. This will ensure that using both headers will always result in a querying the keychain with the `kSecUseDataProtectionKeychain` flag set to `true`, now and in the future.

Additionally, explicitly setting `kSecUseDataProtectionKeychain` to `true` only really affects macOS because the key is set to `true` by default. For macOS, setting `kSecUseDataProtectionKeychain` to `true` requires that targets that include the GoogleUtilities framework are signed with a provisioning profile that includes the Keychain Sharing capability.

More details at go/firebase-macos-keychain-popups

<details open>
<summary>Outstanding work (in order of planned completion)</summary>
<br>

- [x] Open this PR with fix
- [x] Experiment with running macCatalyst tests _(Done in 0449ae6. Can't remove it because macCatalyst has same constraints as macOS.)_
- [x] Unblock CI with #76
- [x] Update 7.8.0's change-log entry with note w.r.t. provisioning profile when using `GULKeychain*` and targeting macOS.
- [x] Add top level documentation to `GULKeychainStorage.h` and `GULKeychainUtils.h` regarding macOS-specific constraints.
- [x] Manual testing on SDKs that depend on GoogleUtilities
- [x] Get this PR approved
- [ ] Add a corresponding bug number in PR description after PR is ready for release (no outstanding work, green CI, etc).
- [x] Get this PR merged
- [ ] Send a following PR to bump the podspec to 7.8.0
- [ ] Stage for 7.8.0 release after above PR is merged  
</details>

Fixes

### CODE REVIEW

**Comment by ncooke3** [comment]
Will add a corresponding bug number in PR description after getting PR ready for release.

**Review by paulb777 — APPROVED** [review]
🥳

File: GoogleUtilities/Environment/SecureStorage/GULKeychainUtils.m [hunk 1]
```diff
@@ -22,14 +22,15 @@ @implementation GULKeychainUtils
 
 + (nullable NSData *)getItemWithQuery:(NSDictionary *)query
                                 error:(NSError *_Nullable *_Nullable)outError {
-  NSMutableDictionary *mutableQuery = [query mutableCopy];
+  NSMutableDictionary *mutableGetItemQuery =
+      [[[self class] multiplatformQueryWithQuery:query] mutableCopy];
 
-  mutableQuery[(__bridge id)kSecReturnData] = @YES;
-  mutableQuery[(__bridge id)kSecMatchLimit] = (__bridge id)kSecMatchLimitOne;
+  mutableGetItemQuery[(__bridge id)kSecReturnData] = @YES;
```
Comment (chliangGoogle) [inline GoogleUtilities/Environment/SecureStorage/GULKeychainUtils.m:28]: curious: Why use mutalbeGetItemQuery here

**Review by chliangGoogle — APPROVED** [review]

File: GoogleUtilities/Environment/SecureStorage/GULKeychainUtils.m (hunk 1 above)
Comment (ncooke3) [reply GoogleUtilities/Environment/SecureStorage/GULKeychainUtils.m:28]: I thought it added more context as what the dictionary was being built for. The implementation of `setItem` is slightly more complicated in that the query may be used in different ways.

Comment (chliangGoogle) [reply GoogleUtilities/Environment/SecureStorage/GULKeychainUtils.m:28]: oh nvm, I thought it is an apple API lol

**Pull request merged**